### PR TITLE
Fix username of reply message when using `WEB_DOMAIN` env.

### PR DESCRIPTION
**Pull request by 3846masa**
On v1.2, Mastodon detects account's local-domain from `./xmlns:author/xmlns:uri`.
e.g. )
- https://github.com/tootsuite/mastodon/blob/v1.2/app/services/process_interaction_service.rb#L16-L17
- https://github.com/tootsuite/mastodon/blob/v1.2/app/services/process_feed_service.rb#L271-L277

`./xmlns:author/xmlns:uri` will be created from `RoutingHelper` methods.
c.f.)
- https://github.com/tootsuite/mastodon/blob/v1.2/app/lib/atom_serializer.rb#L19-L23
- https://github.com/tootsuite/mastodon/blob/v1.2/app/lib/tag_manager.rb#L83-L84

Because of that, Mastodon use web-domain as local-domain.
So, username of reply message is mistaken.
(user@local.example.com (correct) -> user@web.example.com (wrong))

We should replace domain name in `./xmlns:author/xmlns:uri` when using `WEB_DOMAIN` env.

-----

NOTE:

Mastodon may use `./xmlns:author/xmlns:email` for detecting local-domain (e.g. when fetching user stream atom). It works correctly on v1.2.
e.g.)
- https://github.com/tootsuite/mastodon/blob/v1.2/app/services/fetch_remote_account_service.rb#L22-L29

### CODE REVIEW

File: app/lib/atom_serializer.rb [hunk 1]
```diff
@@ -16,7 +16,7 @@ def render(element)
   def author(account)
     author = Ox::Element.new('author')
 
-    uri = TagManager.instance.uri_for(account)
+    uri = TagManager.instance.uri_for(account).gsub(Rails.configuration.x.web_domain, Rails.configuration.x.local_domain)
```
Comment (nightpool) [inline app/lib/atom_serializer.rb]: this seems like a very bad place to fix this. Why not change uri_for?

Comment (Gargron) [inline app/lib/atom_serializer.rb]: Indeed, this is the wrong place

**Review by Gargron — CHANGES_REQUESTED** [review]

**Comment by Gargron** [comment]
>Mastodon may use ./xmlns:author/xmlns:email for detecting local-domain (e.g. when fetching user stream atom). It works correctly on v1.2.

I think Mastodon is the only software which uses the `<email />` tag for this, sadly.

File: app/lib/atom_serializer.rb (hunk 1 above)
Comment (3846masa) [inline app/lib/atom_serializer.rb]: I agree.
However, I don't know where does mastodon use `'TagManager.instance.uri_for`.
So, I can't guess where will changing `TagManager.instance.uri_for` affect.

Should I move that to `uri_for`?
If `TagManager.instance.uri_for` will be changed, I insert `gsub` in [here](https://github.com/tootsuite/mastodon/blob/v1.2/app/lib/tag_manager.rb#L83-L84).

**Comment by 3846masa** [comment]
I agree with @nightpool and @Gargron .
I move that to `uri_for`. Please check it.

**Comment by ThibG** [comment]
Yes, this is what I was debugging yesterday. Unfortunately, changing the URI in those requests does not seem like a correct solution to me, as the users' URI *is* the web part (this is so that mentioned links of unknown users will still go to their web profile — and because that's how Mastodon currently discovers unknown remote mentioned accounts).

More generally, I think we should establish clear rules for mapping account handles (user@domain) with URIs, and use the same rules everywhere instead of doing slightly different things in different places (e.g. salmon requests processing, mentions processing, and fetching a remote user's profile), see #2012 .

**Pull request closed without merging**